src/tools/gcc.jam

```diff
@@ -728,6 +728,12 @@ toolset.flags gcc.compile OPTIONS <local-visibility>global : -fvisibility=defaul
 
 toolset.flags gcc.compile.c++ OPTIONS <rtti>off : -fno-rtti ;
 toolset.flags gcc.compile.c++ OPTIONS <exception-handling>off : -fno-exceptions ;
+toolset.flags gcc.compile.c++ OPTIONS <address-sanitizer>on : -fsanitize=address ;
+toolset.flags gcc.compile.c++ OPTIONS <undefined-sanitizer>on : -fsanitize=undefined ;
+toolset.flags gcc.compile.c++ OPTIONS <thread-sanitizer>on : -fsanitize=thread ;
+toolset.flags gcc.compile.c++ OPTIONS <address-sanitizer>norecover : -fsanitize=address -fno-sanitize-recover=address ;
+toolset.flags gcc.compile.c++ OPTIONS <undefined-sanitizer>norecover : -fsanitize=undefined -fno-sanitize-recover=undefined ;
```

Comment:
The description says "for gcc and clang", so... per clang documentation on this issue:

https://clang.llvm.org/docs/AddressSanitizer.html

To get a reasonable performance add -O1 or higher. To get nicer stack traces in error messages add -fno-omit-frame-pointer. To get perfect stack traces you may need to disable inlining (just use -O1) and tail call elimination (-fno-optimize-sibling-calls).

I don't ususally add -O1, but I do add -g and -fno-omit-frame-pointer, and never tried -fno-optimize-sibling-calls.